File: library/swift/src/StreamClientImpl.swift
import Foundation

/// Envoy implementation of StreamClient.
@ojcMembers
final class StreamClientImpl: NSObject {
  private let engine: EnvoyEngine

  init(engine: EnvoyEngine) {
    self.engine = engine
    super.init()
  }
}

extension StreamClientImpl: StreamClient {
  func newStreamPrototype() -> StreamPrototype {
    return StreamPrototype(engine: self.engine)
  }
}

File: library/kotlin/src/io/envoyproxy/envoymobile/StreamClientImpl.kt
package io.envoyproxy.envoymobile

import io.envoyproxy.envoymobile.engine.EnvoyEngine

/**
 * Envoy's implementation of `StreamClient`.
 */
internal class StreamClientImpl constructor(
  internal val engine: EnvoyEngine
) : StreamClient {

  override fun newStreamPrototype() = StreamPrototype(engine)
}
